src/main/haskell/kore/src/Kore/Step/Simplification/Equals.hs

```diff
@@ -676,12 +676,24 @@ makeEvaluateTermsToPredicateSubstitution
                     axiomIdToSimplfier
                     second
             let
+                toPredicateSafe
```

Comment:
Do you think this definition should be extracted?

Reply:
I think this is a temporary hack (hence the error below) and we should replace it with something else. I think that we will not be able to replace it with another function, we actually need to rewrite the caller. As such, I feel uncomfortable making a library function and hiding that error inside of it, I would prefer to have the error where we need to do the actual fix.